added aten.log_normal_ decomp

min-jean-cho opened the pull request:
Fixes #91275


cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

min-jean-cho committed:
added aten.log_normal_ decomp

min-jean-cho committed:
add log_normal decomp

min-jean-cho commented on torch/_refs/__init__.py:
```diff
@@ -5087,6 +5087,17 @@ def bucketize(
+)
+def log_normal(self, mean=1, std=2, generator=None):
+    assert generator is None
+    return torch.exp(std * torch.randn_like(self) + mean)
```
I wonder if special care is needed to handle overflow, underflow for `log_normal`. For fp32, `torch.exp(x)` overflows for `x >= 89`. 

`torch.randn` practically won't cause this overflow as 99% of values will be in [-3, 3].
But for some `mean`, `std`, this will cause overflow.  
```
>>> torch.empty(100000).log_normal_(mean=89).max()
tensor(inf)
```
I wonder if this behavior is expected in `log_normal_` eager mode.

```
>>> np.random.lognormal(89, 1, 100000).max()
2.9258184949413217e+40
```

lezcano replied:
I think this is alright, but thank you for bringing it up!

lezcano reviewed (commented):
Could you add the relevant OpInfo for this operation as well? It won't be good enough to test the correctness of the implementation, but it will test that the type casting, the out variant and so on are there.

min-jean-cho committed:
(1) added opinfo (2) dtype, parameter check in decomp ref

min-jean-cho committed:
Merge branch 'master' into minjean/inductor_decompose_aten_log_normal

min-jean-cho committed:
opinfo log_normal

min-jean-cho committed:
Merge branch 'minjean/inductor_decompose_aten_log_normal' of https://github.com/min-jean-cho/pytorch into minjean/inductor_decompose_aten_log_normal

min-jean-cho committed:
Merge branch 'pytorch:master' into minjean/inductor_decompose_aten_log_normal

lezcano reviewed (approved):
LGTM provided the tests pass!

min-jean-cho committed:
fix ci

min-jean-cho committed:
lint

min-jean-cho committed:
fix ci

min-jean-cho commented:
[Log-normal] Hi all, thank you for the review. I see all tests have passed now - I've made few minor changes since the last approvals, could you help have a final scan ?

min-jean-cho committed:
Merge branch 'master' into minjean/inductor_decompose_aten_log_normal

min-jean-cho committed:
remove merge conflict mistake

min-jean-cho committed:
lint

min-jean-cho commented:
I will go ahead and merge this as the minor changes after the last approvals are necessary (same changes as https://github.com/pytorch/pytorch/pull/92047)

min-jean-cho commented:
@pytorchbot merge
